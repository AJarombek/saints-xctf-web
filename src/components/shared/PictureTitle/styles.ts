/**
 * JSS styles for the PictureTitle component.
 * @author Andrew Jarombek
 * @since 9/8/2020
 */

import Colors from "../../../styles/colors";
import {FontMixins} from "../../../styles/mixins";

export default {
    container: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center'
    },
    picture: {
        width: 150,
        height: 150,
        borderRadius: '50%',
        backgroundColor: Colors.spotPaletteCream,
        border: '3px solid #888',
        transition: 'border 0.5s ease',

        '&:hover': {
            border: '3px solid #333'
        },

        '& > img': {
            width: '100%',
            height: '100%',
            borderRadius: '50%'
        }
    },
    title: {
        ...FontMixins.robotoSlabBold(),
        fontSize: 20
    },
    subTitle: {
        ...FontMixins.robotoBold()
    }
};
