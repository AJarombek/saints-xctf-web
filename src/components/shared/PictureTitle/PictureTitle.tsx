/**
 * Component for on the profile and group pages which displays a picture (profile picture, group picture) and the
 * title of the page (user's first and last name, group name).
 * You are always very loved.
 * @author Andrew Jarombek
 * @since 9/8/2020
 */

import React from 'react';
import { createUseStyles } from 'react-jss';
import styles from './styles';
import classNames from 'classnames';

interface Props {
  imageUrl: string;
  title: string;
  subTitle: string;
}

const useStyles = createUseStyles(styles);

const PictureTitle: React.FunctionComponent<Props> = ({ imageUrl, title, subTitle }) => {
  const classes = useStyles();

  return (
    <div className={classNames(classes.container, 'pictureTitleContainer')}>
      <figure className={classes.picture}>
        <img src={imageUrl} alt="" />
      </figure>
      <h5 className={classes.title} data-cypress="pictureTitle">
        {title}
      </h5>
      <div className={classes.subTitle} data-cypress="pictureSubTitle">
        {subTitle}
      </div>
    </div>
  );
};

export default PictureTitle;
