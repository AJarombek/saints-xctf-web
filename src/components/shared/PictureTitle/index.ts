import PictureTitle from "./PictureTitle";

export default PictureTitle;
