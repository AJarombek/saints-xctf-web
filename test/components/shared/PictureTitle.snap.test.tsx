/**
 * Snapshot test for the PaginationBar component.
 * @author Andrew Jarombek
 * @since 3/7/2021
 */

import React from 'react';
import renderer from 'react-test-renderer';
import PictureTitle from '../../../src/components/shared/PictureTitle';

describe('PaginationBar Snapshot Tests', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<PictureTitle imageUrl="saintsxctf.png" title="Andy Jarombek" subTitle="@andy" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
