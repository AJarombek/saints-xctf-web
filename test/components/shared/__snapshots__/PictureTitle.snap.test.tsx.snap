// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationBar Snapshot Tests renders correctly 1`] = `
<div
  className="container-0-1-1"
>
  <figure
    className="picture-0-1-2"
  >
    <img
      alt=""
      src="saintsxctf.png"
    />
  </figure>
  <h5
    className="title-0-1-3"
    data-cypress="pictureTitle"
  >
    Andy Jarombek
  </h5>
  <div
    className="subTitle-0-1-4"
    data-cypress="pictureSubTitle"
  >
    @andy
  </div>
</div>
`;
